pandas/core/reshape/merge.py

```diff
@@ -1671,11 +1671,21 @@ def _right_outer_join(x, y, max_groups):
 }
 
 
+def _convert_array_or_index(arg):
+    """Converts DatetimeArray or DatetimeIndex to numpy array in UTC"""
```

Comment:
this is so opaque

use getattr(arg, ‘_values’, arg)._data

Reply:
How about `extract_array(arg)._data`?

Reply:
Thanks for the simplification. Went with the `getattr` solution.